Add Rubocop Minitest

tagliala opened the pull request:
There are 37 `Minitest/NoAssertions` offenses left.

This is due to the fact that there are mocha expectations instead of "assert".

Example:

```rb
  def test_url_helpers_are_recreated_when_defaults_change
    BooksController.expects(:create_resources_url_helpers!).returns(true).once
    BooksController.send(:defaults, instance_name: 'string', collection_name: 'strings')
  end
```

No asserts here

The solution would be to:
- Disable this cop, but we may miss real offenses
- Add `assert true`

tagliala committed:
Add RuboCop minitest

tagliala committed:
Fix safe offenses

- Minitest/AssertEmpty
- Minitest/AssertEmptyLiteral
- Minitest/AssertIncludes
- Minitest/AssertInstanceOf
- Minitest/AssertPredicate
- Minitest/EmptyLineBeforeAssertionMethods
- Minitest/LiteralAsActualArgument

tagliala committed:
Fix unsafe offenses

- Minitest/AssertPredicate
- Minitest/AssertTruthy
- Minitest/RefuteFalse
- Minitest/RefuteIncludes
- Minitest/RefuteRespondTo

tagliala committed:
Disable Minitest/MultipleAssertions cop

javierjulio commented:
@tagliala thanks! I think it would be best to leave the cop on for future changes. Would it be simple enough to add asserts for each of those warnings?

javierjulio commented:
@tagliala forgot that another option would be to just add disable comments for those type of tests which I'd also be fine with. I just think it would be best to leave the cop enabled.

tagliala committed:
Disable Minitest/NoAssertions in tests using mocha

Some tests are using Mocha `expects` and therefore do not require
`assert`.

This commit disables offense on those specific tests, leaving it enabled
for all other ones

tagliala commented:
I've disabled the cop when it is not required: 4eed462cb14117b9eb403e7abd7059abb163b45f

javierjulio reviewed (approved):
Thanks!
